GVFS/GVFS.Common/Maintenance/GitMaintenanceStep.cs

```diff
@@ -143,6 +147,37 @@ protected EventMetadata CreateEventMetadata(Exception e = null)
             return metadata;
         }
 
+        protected bool EnoughTimeBetweenRuns()
+        {
+            if (!this.Context.FileSystem.FileExists(this.TimeFilePath))
+            {
+                return true;
+            }
+
+            string lastRunTime = this.Context.FileSystem.ReadAllText(this.TimeFilePath);
+            DateTime lastRunDateTime = Convert.ToDateTime(lastRunTime);
+            if (DateTime.Now.Subtract(lastRunDateTime) >= this.TimeBetweenRuns)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        protected void RecordRunTime()
+        {
+            if (!this.Context.FileSystem.TryWriteTempFileAndRename(this.TimeFilePath, DateTime.Now.ToString(), out Exception handledException))
+            {
+                this.Context.Tracer.RelatedError("Failed record run time: {0}", handledException?.ToString());
+            }
+        }
+
+        protected IEnumerable<int> RunningGitProcessIds()

```

Comment:
We print the results for debugging purposes:
                            activity.RelatedWarning("Skipping LooseObjectsStep due to git pids {0}", string.Join(",", processIds));

We could add debugging on this method for the processIds and then another warning line in the LooseObjectStep (to confirm that's why we're exiting).

I perfer it as is honestly.

Reply:
I got this warning as I was running it on my machine:

```
[2018-12-14 10:15:46 -05:00] LooseObjectsStep (Start)
[2018-12-14 10:15:46 -05:00] Warning {"WarningMessage":"Skipping LooseObjectsStep due to git pids 1808"}
[2018-12-14 10:15:46 -05:00] LooseObjectsStep (Stop) {"DurationMs":18}
```

This seems helpful, especially if we start seeing users who never get the work done because they always have processes running. We can monitor via Kusto for user who always hit this warning instead of an actual job run.

Reply:
This brings up a great point.  I'm currently not logging to telemetry.  We should, do that but carefully!

@derrickstolee I'm thinking if the time span has elapsed and we bail for active pids, that would be the only time we want a warning in telemetry.  Agree?